scripts/build_changed_modules.py

```diff
@@ -0,0 +1,56 @@
+# This script test builds the updated modules for a pull request branch.
+
+import json
+import os
+import subprocess
+import argparse
+
+module_registry_file = 'modules.json'
+
+def get_modules():
+  if not os.path.isfile(module_registry_file):
+    return {}
+  f = open(module_registry_file, 'r')
+  modules = json.load(f)
+  f.close()
+  return modules
+
+def checkout(commit):
+  args = ['git', 'checkout', commit]
+  print(" ".join(args))
+  subprocess.run(args)
+
+def build_module(module_id):
+  args = ['nix', 'build', '.#modules."%s"' % module_id]
+  print(" ".join(args))
+  subprocess.run(args)
+
+def main():
+  parser = argparse.ArgumentParser(
+    prog='build_changed_modules',
+    description='builds the modules that were changed from an upstream branch',
+  )
+
+  parser.add_argument('upstream_branch')
+  parser.add_argument('current_branch')
+
+  args = parser.parse_args()
+
+  checkout(args.upstream_branch)
+  upstream_modules = get_modules()
```

Comment:
you can `git show <rev> modules.json` - better imo since it doesn't mutate the environment